dspace-api/src/main/java/org/dspace/discovery/configuration/DiscoveryConfiguration.java

```diff
@@ -51,6 +51,15 @@
     private DiscoveryHitHighlightingConfiguration hitHighlightingConfiguration;
     private DiscoveryMoreLikeThisConfiguration moreLikeThisConfiguration;
     private boolean spellCheckEnabled;
+    private boolean indexAlways = false;
+
+    public boolean isIndexAlways() {
```

Comment:
Please add Javadocs.  I'm not sure I understand this configuration and when something would be "true" vs "false".  Does "false" mean something is *not* indexed? or only indexed occasionally?  I think we should have more specific examples of usage in the Javadocs here -- I'm having difficulty understanding this property myself and why it's needed only on Entities configurations, and not on others (like `workflowConfiguration`, `workspaceConfiguration`, etc)

Reply:
The Javadocs have been included.
The `workflowConfiguration` and `workspaceConfiguration` currently use custom code in the SearchUtils and SolrServiceImpl class to override the behavior. Indexing workspace and workflow items is still planned to be improved
This PR is focussed on entities configurations, and avoids creating custom code in SearchUtils and SolrServiceImpl with a hard-coded list of potential entity types